sources/api/shibaken/src/main.rs

```diff
@@ -0,0 +1,279 @@
+/*!
+# Introduction
+
+shibaken is called by sundog as a setting generator.
+
+shibaken will fetch and populate the admin container's user-data with authorized ssh keys from the
+AWS instance metadata service (IMDS).
+
+(The name "shibaken" comes from the fact that Shiba are small, but agile, hunting dogs.)
+*/
+
+#![deny(rust_2018_idioms)]
+
+use log::{debug, info, warn};
+use reqwest::blocking::Client;
+use serde::Serialize;
+use simplelog::{Config as LogConfig, LevelFilter, TermLogger, TerminalMode};
+use snafu::ResultExt;
+use std::str::FromStr;
+use std::{env, process};
+
+// Instance Meta Data Service.
+//
+// Currently only able to get fetch session tokens from `latest`
+// FIXME Pin to a date version that supports IMDSv2 once such a date version is available.
+const IMDS_PUBLIC_KEY_BASE_URL: &str = "http://169.254.169.254/latest/meta-data/public-keys";
+const IMDS_SESSION_TOKEN_ENDPOINT: &str = "http://169.254.169.254/latest/api/token";
```

Comment:
```suggestion
const IMDS_SESSION_TOKEN_URI: &str = "http://169.254.169.254/latest/api/token";
```
To match changes below.